readthedocs/builds/forms.py

```diff
@@ -37,3 +39,77 @@ def save(self, commit=True):
         if obj.active and not obj.built and not obj.uploaded:
             trigger_build(project=obj.project, version=obj)
         return obj
+
+
+class RegexAutomationRuleForm(forms.ModelForm):
+
+    ALL_VERSIONS_REGEX = r'.*'
+    SEMVER_REGEX = r'^v?(\d+\.)(\d+\.)(\d)(-.+)?$'
+    MATCH_CHOICES = (
+        (ALL_VERSIONS_REGEX, 'All versions'),
+        (SEMVER_REGEX, 'Semver versions'),
+        (None, 'Custom'),
+    )
+
+    match = forms.ChoiceField(
+        label='Rule',
+        choices=MATCH_CHOICES,
+        initial=ALL_VERSIONS_REGEX,
+        required=False,
+    )
+
+    match_arg = forms.CharField(
+        label='Custom rule',
+        help_text=_('A Python regular expression'),
+        required=False,
+    )
+
+    class Meta:
+        model = RegexAutomationRule
+        fields = ['description', 'match', 'match_arg', 'version_type', 'action']
+
+    def __init__(self, *args, **kwargs):
+        self.project = kwargs.pop('project', None)
+        super().__init__(*args, **kwargs)
+
+        self.fields['version_type'].choices = [
+            (None, '-' * 9),
+            (BRANCH, BRANCH_TEXT),
+            (TAG, TAG_TEXT),
+        ]
+
+        match_options = set(v[0] for v in self.MATCH_CHOICES)
+        if self.instance.pk:
+            match_arg = self.instance.match_arg
+            if match_arg and match_arg in match_options:
+                self.initial['match'] = self.instance.match_arg
+            else:
+                self.initial['match'] = None
+
+    def clean_match_arg(self):
+        match_arg = self.cleaned_data['match_arg']
+        match = self.cleaned_data['match']
+        if match:
+            match_arg = match
+        if not match_arg:
+            raise forms.ValidationError(
+                _('Custom match should not be empty.'),
+            )
+        return match_arg
+
+    def clean_description(self):
+        description = self.cleaned_data['description']
+        if not description:
+            description = self.instance.get_description()
```

Comment:
I think it's better to leave it blank if the user does not enter any description and make this check when rendering. This way we are not duplicating the same value over and over in the db and also if we change it later with a better message, it will change for all the users.

Reply:
But then the user would see the description in list page and not on the form, I think that would be confusing

Reply:
I'm fine with that.

Although, if you want to show it in the form as well, you can add it in the `Form.__init__` method.

Reply:
We can't do that bc we don't know what option is going to be selected yet

Reply:
The thing is that you want to populate that field with the default text _only_ when editing the rule. When creating the rule, the field should be empty so the user can write anything they want.

Reply:
Is empty by default, if the form is saved without  a description, the rule is saved with the default description, otherwise use what the user wrote

Reply:
Right. Why do we want to save the same value over and over again in the database if it's the default when it's `None`? That's my point.